### PR TITLE
Migration to Spring 4

### PR DESCRIPTION
**Pull request by buehner**
This simple pull request migrates SHOGun to Spring 4 and Spring Security 4.

We are currently using the spring-4 branch of SHOGun in a (big) project and everything seems to be fine.

Would be great if someone could review and test this branch with another application before merging.

http://docs.spring.io/spring-security/site/migrate/current/3-to-4/html5/migrate-3-to-4-xml.html

:beers:

### CODE REVIEW

**Comment by ahennr** [comment]
THX @buehner .

I used this branch in my application as well.
As far as I can see, updating Spring (3.2.11.RELEASE -> 4.2.3.RELEASE) and Spring Security (3.2.5.RELEASE -> 4.0.3.RELEASE) didn't conflict with your changes.

Nevertheless, IMO the Spring config files (especially `applicationContext-security.xml`) also need to be updated as well, since there are some changes mentioned in the migration guide. Or am I wrong at this point?

**Comment by buehner** [comment]
Thx for your fixes @ahennr !

I just created a new releasehttps://github.com/terrestris/shogun/releases/tag/v0.3 for people that still rely on the current state of SHOGun.

I think it would be great, if @marcjansen and @chrismayer could have a quick look at this before merging this.

File: src/main/webapp/WEB-INF/spring/db-config.xml [hunk 1]
```diff
@@ -77,12 +77,9 @@
...
 	<bean id="sessionFactory"
 		class="org.springframework.orm.hibernate3.annotation.AnnotationSessionFactoryBean">
-		<property name="dataSource">
-			<ref local="dataSource" />
-		</property>
-		<property name="packagesToScan">
-			<ref local="dbEntitiesPackages"></ref>
-		</property>
+		<property name="dataSource" ref="dataSource" />
+        <property name="packagesToScan" ref="dbEntitiesPackages" />
```
Comment (marcjansen) [inline src/main/webapp/WEB-INF/spring/db-config.xml]: Indentation is off by one tab.

File: src/main/webapp/WEB-INF/spring/initialize-beans.xml [hunk 2]
```diff
@@ -138,23 +138,13 @@
 			</list>
 		</property>
 
-		<property name="mapConfigs">
-			<list>
-				<ref local="stdMapConfig" />
-			</list>
-		</property>
+		<property name="mapConfigs" ref="stdMapConfig" />
```
Comment (marcjansen) [inline src/main/webapp/WEB-INF/spring/initialize-beans.xml]: This was a list previously. Will this work the new way?

**Comment by marcjansen** [comment]
I have added two comments, otherwise this looks good to me. I am :+1: for merging.

Please give @chrismayer at least 2 days to give his :+1: or :-1:.

File: src/main/webapp/WEB-INF/spring/db-config.xml (hunk 1 above)
Comment (ahennr) [inline src/main/webapp/WEB-INF/spring/db-config.xml]: fixed in de536a9

File: src/main/webapp/WEB-INF/spring/initialize-beans.xml (hunk 2 above)
Comment (ahennr) [inline src/main/webapp/WEB-INF/spring/initialize-beans.xml]: No it won't. I erroneously removed the definition if the list, sry. :confounded:

Reverted this in 59385c0

**Comment by buehner** [comment]
Thx @ahennr .
Let's wait a bit more for @chrismayer before merging.

**Comment by chrismayer** [comment]
Hi all,
the changesets are looking good to me. I am :+1: for merging this.
Thanks for tackling this!

**Comment by ahennr** [comment]
Will merge now

**Comment by marcjansen** [comment]
Thanks for the work and the reviews, gentlemen.

**Pull request merged**